src/MIDebugEngine/Engine.Impl/Breakpoints.cs

```diff
@@ -329,6 +377,12 @@ internal BoundBreakpoint(PendingBreakpoint parent, ulong addr, /*optional*/ Tupl
             }
         }
 
+        internal BoundBreakpoint(PendingBreakpoint parent, ulong addr, uint size)
+        {
+            Addr = addr;
```

Comment:
Now uses IsDataBreakpoint to distinguish the difference, Also BreakpointManager explicitly deals with code and data breaks separately.